Fix dirfd cache and apply other patches from Gentoo Linux

mgorny opened the pull request:
<!--
Thank you for your pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->

Fixes #11

<!--
Please add any other relevant info below:
-->

Grab the whole set of patches from Gentoo Linux that fix a few CVEs and some other bugs in `patch(1)`. Notably, this includes a fresh patch that fixes dirfd cache when building with `-DNDEBUG`, that caused `patch(1)` to run out of fds when applying "large" (40 file with `ulimit -n 1024`) patches.

mgorny committed:
Copy patches from Gentoo to resolve multiple issues

Copy the patchset used by Gentoo since that's the least-effort way
of fixing multiple issues with patch-2.7.6.  For me, it fixes
the problem where applying a 40-file patch would open over 1000 fds,
effectively exceeding the default ulimit.  However, this also includes
a number of CVE fixes and other bugs that seem useful.

I have kept the original file names to make syncing easier
in the future.

mgorny committed:
MNT: Re-rendered with conda-build 25.1.1, conda-smithy 3.45.4, and conda-forge-pinning 2025.02.04.19.34.29

mgorny committed:
Adjust Gentoo patches for conda-build compatibility

Split the two commits combined in one patch into two files, since
otherwise `patch --dry-run` fails.  Adjust the "configure-mkdir" patch
for `-p0` since conda-build misdetects the patch level.

mgorny committed:
Add a patch to fix fd caching with `-NDEBUG`

Upstream was inserting cache entries inside an `assert()`, so our
`-DNDEBUG` disabled caching entirely.  Add a patch from Sam James
(@thesamesam) to fix it.  This also fixes the deep-directories test.

Fixes #11

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipe/meta.yaml```) and found some lint.

Here's what I've got...


For **recipe/meta.yaml**:

 * ❌ This recipe is using a compiler, which now requires adding a build dependence on `{{ stdlib("c") }}` as well. Note that this rule applies to each output of the recipe using a compiler. For further details, please see https://github.com/conda-forge/conda-forge.github.io/issues/2102.

For **recipe/meta.yaml**:

 * ℹ️ The recipe is not parsable by parser `conda-forge-tick (the bot)`. Your recipe  may not receive automatic updates and/or may not be compatible with conda-forge's infrastructure. Please check the logs for more information and ensure your recipe can be parsed.
 * ℹ️ The recipe is not parsable by parser `conda-souschef (grayskull)`. This parser is not currently used by conda-forge, but may be in the future. We are collecting information to see which recipes are compatible with grayskull.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/13162335086](https://github.com/conda-forge/conda-forge-webservices/actions/runs/13162335086). Examine the logs at this URL for more detail.</sub>

mgorny committed:
Add stdlib('c')

mgorny committed:
MNT: Re-rendered with conda-build 25.1.2, conda-smithy 3.45.4, and conda-forge-pinning 2025.02.05.19.16.07

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/meta.yaml```) and found it was in an excellent condition.

I do have some suggestions for making it better though...


For **recipe/meta.yaml**:

 * ℹ️ The recipe is not parsable by parser `conda-forge-tick (the bot)`. Your recipe  may not receive automatic updates and/or may not be compatible with conda-forge's infrastructure. Please check the logs for more information and ensure your recipe can be parsed.
 * ℹ️ The recipe is not parsable by parser `conda-souschef (grayskull)`. This parser is not currently used by conda-forge, but may be in the future. We are collecting information to see which recipes are compatible with grayskull.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/13184760370](https://github.com/conda-forge/conda-forge-webservices/actions/runs/13184760370). Examine the logs at this URL for more detail.</sub>

mgorny committed:
Sync patches again

The Gentoo patchset has been updated, so sync up again, while this
has not been merged yet.

mgorny committed:
Combine build/number keys to fix autotick bot

Autotick bot recipe parsers can't handle "duplicate" keys in input YAML,
so instead of using Jinja `{% ... %}` condition blocks, use inline
condition expressions.

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/meta.yaml```) and found it was in an excellent condition.

mgorny commented:
I've also added small refactoring to make the recipe compatible with autotick bot. Hope that's fine — if not, feel free to remove it.

mgorny commented:
Gentle ping.

mgorny commented:
@nehaljwani, do you need help maintaining this feedstock? I'd gladly become a co-maintainer if you agree.

thesamesam commented:
The fix is in the new GNU Patch 2.8 release, FWIW.

h-vetinari commented:
> I'd gladly become a co-maintainer if you agree.

@mgorny, if you add yourself here I'd be happy to merge

mgorny committed:
Add myself as a co-maintainer

Signed-off-by: Michał Górny <mgorny@quansight.com>

mgorny committed:
MNT: Re-rendered with conda-smithy 3.52.1 and conda-forge-pinning 2025.09.09.16.04.35

Other tools:
- conda-build 25.7.0
- rattler-build 0.47.0
- rattler-build-conda-compat 1.4.6

mgorny commented:
Added and rerendered.

h-vetinari committed:
add a mirror

h-vetinari commented:
Thanks! I'd still like to turn
https://github.com/conda-forge/patch-feedstock/blob/843457fe780f65a3f734947f0401c9222e5eea2f/conda-forge.yml#L1-L4

into
```yaml
build_platform:
  linux_aarch64: linux_64
  linux_ppc64le: linux_64
  osx_arm64: osx_64
conda_forge_output_validation: true
github:
```
in order to cross-compile aarch/ppc; but if you want we can do that in a follow-up.

mgorny committed:
Update platforms as requested by @h-vetinari

Signed-off-by: Michał Górny <mgorny@quansight.com>

h-vetinari commented:
@conda-forge-admin, please rerender

mgorny committed:
MNT: Re-rendered with conda-smithy 3.52.1 and conda-forge-pinning 2025.09.09.16.04.35

Other tools:
- conda-build 25.7.0
- rattler-build 0.47.0
- rattler-build-conda-compat 1.4.6

mgorny commented:
Uh, looks like something went wrong with the rerendering service.

h-vetinari commented:
Yeah. And there's also something going on with https://conda.anaconda.org - getting lots of 503s (not just here)

h-vetinari reviewed (approved):
Thanks! As a follow-up, it'd be nice to update to 2.8 as Sam mentioned. :)

mgorny commented:
Thanks. I'll probably go for rattler-build too, for the sake of exercise.
